src/app/dataset-view/additional-components/data-component/data-component.ts

```diff
@@ -51,6 +51,12 @@ export class DataComponent extends BaseComponent implements OnInit {
 
     public ngOnInit(): void {
         this.trackSubscriptions(
+            this.appDatasetSubsService.onDatasetDataSqlErrorOccured.subscribe(
+                (dataSqlErrorUpdate: DataSqlErrorUpdate) => {
+                    this.sqlErrorMarker = dataSqlErrorUpdate.error;
+                    this.cdr.markForCheck();
+                },
+            ),
```

Comment:
I suggest to use pipe `async`.
I it will check for changes and unsubscribe by itself

Reply:
It's a reasonable comment, however it should be done as a separate refactoring issue, and for many locations like this, as this is not the only place for a related improvement

Reply:
Need discuss about it.

Reply:
Aggree

Reply:
Reported https://github.com/kamu-data/kamu-web-ui/issues/129